compiler/vmdef.nim

```diff
@@ -237,6 +237,17 @@ type
     currentLineInfo*: TLineInfo
   VmCallback* = proc (args: VmArgs) {.closure.}
 
+  PStackFrame* = ref TStackFrame
+  TStackFrame* = object
+    prc*: PSym                 # current prc; proc that is evaluated
+    slots*: seq[TFullReg]      # parameters passed to the proc + locals;
+                              # parameters come first
+    next*: PStackFrame         # for stacking
+    comesFrom*: int
+    safePoints*: seq[int]      # used for exception handling
+                              # XXX 'break' should perform cleanup actions
+                              # What does the C backend do for it?
+
```

Comment:
So previously, this was an implementation detail of ``vm.nim``, now it's in ``vmdef.nim``. Seems to be a refactoring that goes in the wrong direction.

Reply:
what do you mean? I didn't add any dependencies/imports (besides innocuous std/private/miscdollars); the only diff is adding `tosSaved*: PStackFrame` in `TCtx`
(and moving PStackFrame, TStackFrame above PCtx but that's just cosmetic and optional)

PStackFrame already was in vmdef (and used in vmprofiler.nim, vm.nim) before PR.

Or are you referring to the addition of `tosSaved` field? if so, what are concrete downsides?

Reply:
I misread the diff, sorry.